### PR TITLE
Update unit tests and simplify logic  for dynamic update

### PR DESCRIPTION
**Pull request by jshum2479**
Russ and I worked on adding unit tests and simplify the makeright logic for dynamic update.

### CODE REVIEW

File: operator/src/test/java/oracle/kubernetes/operator/helpers/PodHelperTestBase.java [hunk 1]
```diff
@@ -1520,6 +1590,13 @@ public void whenPodCreated_createPodWithOwnerReference() {
     assertThat(getCreatedPod().getMetadata().getOwnerReferences(), contains(expectedReference));
   }
 
+  // TODO: add tests for dynamic updates
```
Comment (rjeberhard) [inline operator/src/test/java/oracle/kubernetes/operator/helpers/PodHelperTestBase.java:1593]: How hard would it be to create these tests? I'm not a fan of leaving a TODO in the code.

Comment (jshum2479) [reply operator/src/test/java/oracle/kubernetes/operator/helpers/PodHelperTestBase.java:1593]: actually the tests are there, removing TODO

**Review by rjeberhard — APPROVED** [review]

**Comment by jshum2479** [comment]
There is regression in the branch, withdrawing now.

**Comment by jshum2479** [comment]
I found the problem, actually it's a bug rather than a regression in mii v1 model diff code

**Review by russgold — APPROVED** [review]

**Comment by jshum2479** [comment]
@rjeberhard  Actually, it's not a regression, I found a bug in v1 and fixed that.  It should be good to go

**Pull request merged**